homeassistant/components/sma/__init__.py

```diff
@@ -50,7 +52,9 @@ def _parse_legacy_options(entry: ConfigEntry, sensor_def: pysma.Sensors) -> list
     # Add sensors from the custom config
     sensor_def.add(
         [
-            pysma.Sensor(o[CONF_KEY], n, o[CONF_UNIT], o[CONF_FACTOR], o.get(CONF_PATH))
+            pysma.sensor.Sensor(
+                o[CONF_KEY], n, o[CONF_UNIT], o[CONF_FACTOR], o.get(CONF_PATH)
+            )
```

Comment:
This will add a metric ton of sensors to Home Assistant, most of which users don't need.

See the documentation PR for an mild idea:

<https://github.com/home-assistant/home-assistant.io/pull/17987/files>

What should have happened in this PR, is that less common sensors should have been disabled in the entity registry by default.

Please adjust this in a follow-up PR.